src/service.coffee

```diff
@@ -145,14 +139,25 @@ module.exports = (settings)->
 
     es.random options
 
-  service.get '/:type/:id' , jsonP( (req)->
+  service.get '/:type/:id' , jsonP( (req, res, next) ->
     tf = settings.types[req.params.type]?.documentTransform
 
-    es.fetch(req.params.id,req.params.type).then (body)->
-      if tf? and tf.transform? then tf.transform body._source else body._source
+    es.fetch req.params.id, req.params.type
+      .then (body) ->
+        if tf? and tf.transform? then tf.transform body._source else body._source
+      .catch (error) ->
+        if settings.notFoundHandlers?
+          next()
+        else
+          console.error error
+          error
   )
 
+  if settings.notFoundHandlers?
+    for handler in settings.notFoundHandlers
+      if typeof handler.install == 'function'
+        handler.install service
+      else
+        console.error 'function install missing in "not found handler"'
```

Comment:
If this is an error, we should throw an exception, no?
OTOH, I see I did it the same way in line 49ff. Never mind.